File: server/src/main/java/com/finale/neulhaerang/domain/routine/dto/response/RoutineResDto.java
package com.finale.neulhaerang.domain.routine.dto.response;

import java.time.LocalTime;
import java.util.List;

import com.finale.neulhaerang.domain.routine.entity.Routine;
import com.finale.neulhaerang.domain.routine.entity.StatType;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class RoutineResDto {
	private Long routineId;
	private String content;
	private StatType statType;
	private boolean alarm;
	private LocalTime alarmTime;
	private List<Boolean> repeated;

	public static RoutineResDto of(Routine routine, List<Boolean> repeated) {
		return RoutineResDto.builder()
			.routineId(routine.getId())
			.content(routine.getContent())
			.alarm(routine.isAlarm())
			.alarmTime(routine.getAlarmTime())
			.statType(routine.getStatType())
			.repeated(repeated)
			.build();
	}
}

File: android/data/src/main/java/com/finale/neulhaerang/data/model/response/RoutineResDto.kt
package com.finale.neulhaerang.data.model.response

import java.time.LocalTime

/**
 * 루틴 조회 응답
 * @param dailyRoutineId
 * @param routineId
 * @param content
 * @param check 완료 여부
 * @param alarm
 * @param alarmTime
 * @param statType String
 * @param repeated
 */
data class RoutineResDto(
    val dailyRoutineId: Long = 0,
    val routineId: Long,
    val content: String,
    val check: Boolean = false,
    val alarm: Boolean = false,
    val alarmTime: LocalTime?,// = LocalTime.now(),
    val statType: String = "갓생력",
    val repeated: List<Boolean> = List(7) { _ -> false },
)